pkg/types/azure/validation/platform.go

```diff
@@ -0,0 +1,15 @@
+package validation
+
+import (
+	"github.com/openshift/installer/pkg/types/azure"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidatePlatform checks that the specified platform is valid.
+func ValidatePlatform(p *azure.Platform, fldPath *field.Path) field.ErrorList {
```

Comment:
shouldn't this include that region and basedomainresourcegroup be set?

Reply:
added validation and tests